### PR TITLE
USWDS-Next - Storybook: Add initial implementation

### LINKED ISSUE #6: Configure Storybook

### PR DESCRIPTION
**Pull request by mejiaj**
Closes #6.

Blocked by #8.

Initial setup of StorybookJS in repo. Includes a few examples for:

- Custom docs with MDX.
- Embedding existing MD content.
- Autogenerated docs.
- Component previews.

Also created a small custom theme called `UswdsTheme.js` to use Public Sans, bump up default font size, and use some USWDS colors.

**Welcome page**
Includes an example of what custom documentation using [Storybook blocks](https://storybook.js.org/docs/writing-docs/doc-blocks) looks like. 
image

**Embedded README**
Example of embedding an existing README into page. Could be used for CHANGELOGs.
image

**Component docs**
Autogenerated component docs.
image

**Component preview**
Example of importing Lit component with configurable props.
image

## How to test

1. Go into `web-components` directory.
2. Run `npm install`.
3. Run `npm start`.

You should have a working storybook library powered by Vite.

### CODE REVIEW

File: web-components/src/components/usa-link/link.stories.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import "./index";
+
+import { html } from 'lit';
```
Comment (mahoneycm) [inline web-components/src/components/usa-link/link.stories.js]: Use double quotes for consistency
```suggestion
import { html } from "lit";
```

**Comment by mejiaj** [comment]
## Docs update
I've enabled source code by default [0523b65] to avoid having to click on each variant for markup.

| Before | After |
|--------|--------|
| image | image |

**Review by heymatthenry — APPROVED** [review]
Thanks for putting this all together! This will be a _huge_ help.

File: web-components/.storybook/preview.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      matchers: {
+        color: /(background|color)$/i,
+        date: /Date$/i,
+      },
+    },
+    docs: {
+      toc: true, // Autogenerate table of contents.
+      theme: UswdsTheme,
+      canvas: {
+        sourceState: 'shown'
```
Comment (mejiaj) [inline web-components/.storybook/preview.js]: > [!NOTE]  
> **Not a blocker**, but indentation can be off in some previews. There's a workaround in this issue [DocsPage code preview indentation is wrong · Issue #8078 · storybookjs/storybook](https://github.com/storybookjs/storybook/issues/8078).

**Review by mahoneycm — APPROVED** [review]
Thanks so much for doing this! Pulled it into the card component work and it's working like a charm. One small comment for quote consistency on the stories page but not a blocker

File: web-components/src/components/usa-link/link.stories.js (hunk 1 above)
Comment (mejiaj) [reply web-components/src/components/usa-link/link.stories.js]: Good point. Fixed in 41ec42b.

File: web-components/.storybook/preview.js (hunk 2 above)
Comment (caseywatts) [reply web-components/.storybook/preview.js]: FYI, I got this working on the MD WDS project I'm working on!

Here are my notes on the workaround:
https://github.com/storybookjs/storybook/issues/8078#issuecomment-2325332120

Comment (mejiaj) [reply web-components/.storybook/preview.js]: @caseywatts thanks for sharing. Created an issue to give this a try 😄

**Pull request merged**